### PR TITLE
Issue #162 - High level view of definition (icon and empty component)

### LINKED ISSUE #162: High level view of definitions
Adding a component to the list should show users a high level view of the definition that includes:

- indications of type, provider, namespace, name, and version
- ClearlyDefined score
- Declared SPDX License expression
- Indicators as to whether or not the entry has any data at all in the system, been curated, or has outstanding contributions

### PR DESCRIPTION
**Pull request by storrisi**
Fixes #162 , completing the outstanding implementation for that issue

### CODE REVIEW

File: src/components/ComponentList.js [hunk 1]
```diff
@@ -104,25 +104,35 @@ export default class ComponentList extends React.Component {
...
+            !readOnly &&
+            this.renderButtonWithTip(
+              <Button onClick={this.addSourceForComponent.bind(this, component)}>
+                <i className="fas fa-code" />
+              </Button>,
+              'Add the definition for source that matches this package'
             )}
           {this.renderButtonWithTip(
-            <Button className="list-fa-button" onClick={this.inspectComponent.bind(this, currentComponent, definition)}>
+            <Button
```
Comment (jeffmcaffer) [inline src/components/ComponentList.js]: better here would be to only show the "add source" button if there is actually a sourcelocation in the given definition. That works for the empty case as well as improving the experience when there is a definition and it just happens to not have a source location.

Also means (I think) that you can get rid of the whole "disabled" concept.

File: src/components/TwoLineEntry.js [hunk 2]
```diff
@@ -13,18 +13,19 @@ export default class TwoLineEntry extends React.Component {
...
     onClick: () => {}
   }
 
   render() {
-    const { buttons, image, headline, message, onClick, letter, panel, highlight } = this.props
+    const { buttons, image, headline, message, onClick, letter, panel, highlight, isDisabled } = this.props
     return (
       <div className="two-line-entry">
-        <div className="list-row" onClick={onClick}>
+        <div className={`list-row${isDisabled ? ' isDisabled' : ''}`} onClick={onClick}>
```
Comment (jeffmcaffer) [inline src/components/TwoLineEntry.js]: suggest we change from "disabled" to "empty" or some such. We might actually want the user to be able to do some things on empty entries (e.g., in theory they could start adding curations)

File: src/utils/definition.js [hunk 3]
```diff
@@ -30,4 +30,14 @@ export default class Definition {
...
     return { tool, effective }
   }
+
+  /**
+   * Determine if a component doesn't have any data. In order to show in the UI in the list of Components
+   *
+   * @param {*} definition
+   * @returns {boolean}
+   */
+  static isDefinitionEmpty(definition) {
```
Comment (jeffmcaffer) [inline src/utils/definition.js]: This naming and commenting is too broad. There could be tons of other data, and just the source location is missing (actually that is the 90% case).

**Comment by storrisi** [comment]
@jeffmcaffer this is how the implementation looks like:
sep-21-2018 17-28-58

File: src/components/ComponentList.js (hunk 1 above)
Comment (storrisi) [reply src/components/ComponentList.js]: @jeffmcaffer i've updated the code based on your suggestion.

This is what is gonna happen if the definition has an empty sourceLocation:
schermata 2018-09-28 alle 09 31 19

Comment (storrisi) [reply src/components/ComponentList.js]: @jeffmcaffer what should be the best way to check if a definition is empty?

Actually, a definition that hasn't any data, has a structure like this:
schermata 2018-09-28 alle 15 30 51

Given that, we checked for the existence of the field "sourceLocation".
That's why, from our perspective, the best way to check an empty definition was to check the sourceLocation field.

**Comment by storrisi** [comment]
I don't know why you can't see my comments.
By the way i've improved the method that check if a definition is empty, watching for the `described.tools` array field

File: src/utils/definition.js [hunk 4]
```diff
@@ -32,6 +33,20 @@ export default class Definition {
...
+    return !definition.described || (definition.described && isEmpty(definition.described.tools))
+  }
+
+  static isSourceEmpty(definition) {
+    return !definition.described || (definition.described && definition.described.sourceLocation === undefined)
```
Comment (jeffmcaffer) [inline src/utils/definition.js]: Simplify to `return !!get(definition, 'described.sourceLocation')`

File: src/utils/definition.js [hunk 5]
```diff
@@ -32,6 +33,20 @@ export default class Definition {
...
   }
 
+  /**
+   * Determine if a component doesn't have any data. In order to show in the UI in the list of Components
+   *
+   * @param {*} definition
+   * @returns {boolean}
+   */
+  static isDefinitionEmpty(definition) {
+    return !definition.described || (definition.described && isEmpty(definition.described.tools))
```
Comment (jeffmcaffer) [inline src/utils/definition.js]: simplify. `return isEmpty(get(definition, 'described.tools'))`

**Pull request merged**